README.md

```diff
@@ -122,6 +122,172 @@ your container. This will allow you to login to the root account remotely. Local
+ADD 60_verify_content_admin.sh /usr/share/container-layer/mysql/pre-init/
+ADD 60_usage_content_admin.txt /usr/share/container-layer/mysql/usage/
+USER 27
+```
+
+Content of the `60_content_admin_create.sh` file
+```
+mysql $mysql_flags <<EOSQL
+    CREATE USER '${MYSQL_CONTENT_ADMIN_USER}'@'%' IDENTIFIED BY '${MYSQL_CONTENT_ADMIN_PASSWORD}';
+    GRANT INSERT, SELECT, UPDATE, DELETE ON \`${MYSQL_DATABASE}\`.* TO '${MYSQL_USER}'@'%' ;
+    FLUSH PRIVILEGES ;
+EOSQL
+```
+
+Content of the `60_content_admin_password.sh` file
+```
+mysql $mysql_flags <<EOSQL
+  SET PASSWORD FOR '${MYSQL_CONTENT_ADMIN_USER}'@'%' = PASSWORD('${MYSQL_CONTENT_ADMIN_PASSWORD}');
+EOSQL
+```
+
+Content of the `60_verify_content_admin.sh` file:
+```
+if ! [[ -v MYSQL_CONTENT_ADMIN_USER && -v MYSQL_CONTENT_ADMIN_PASSWORD ]]; then
+    usage
+fi
+
+[ ${#MYSQL_CONTENT_ADMIN_USER} -le 16 ] || usage "MySQL username for MYSQL_CONTENT_ADMIN_USER too long (maximum 16 characters)"
+```
+
+Content of the `60_usage_content_admin.txt` file:
+```
+Content admin user:
+  MYSQL_CONTENT_ADMIN_USER (max 16 characters)
+  MYSQL_CONTENT_ADMIN_PASSWORD
+```
+
+To build such a layered Dockerfile, just run `docker build .` in the directory, that includes all the files above. Built Docker image may be then run as:
+
+```
+docker run -ti --rm -e MYSQL_USER=user -e MYSQL_PASSWORD=pass1 -e MYSQL_DATABASE=db -e MYSQL_CONTENT_ADMIN_USER=contentuser -e MYSQL_CONTENT_ADMIN_PASSWORD=pass2 <image>
+```
+
+### Example of removing existing features in layered Dockerfile
+
+There are already some files in the directories mentioned above. These files provide the basic features that are used by majority of users.
+
+Example of such a feature is creation of one database with specified name and user that can work with that database. In order to remove this feature, the files need to be removed in the Dockerfile:
+
+```
+FROM <this-image>
+RUN rm -f /usr/share/container-layer/mysql/post-init/20-base-database.sh \
+/usr/share/container-layer/mysql/pre-start/10-passwords.sh \
+/usr/share/container-layer/mysql/pre-init/10-validate-base-variables.sh
+```
+
+Such an image won't require `MYSQL_USER`, `MYSQL_PASSWORD` and `MYSQL_DATABASE` environment variables specified.
+
+
+Extending the docker image by volume mounting files
+---------------------------------------------------
+
+This Docker image allows to adjust functionality even without creating the layered Dockerfile as described above. It is possible to use volume mounting using `-v` option (see `docker-run` man page for details).
+
+In a nutshell, all we need to do is adding the same files, as we added in examples above, just using volume mounting feature of the `docker utility.
+
+### Example of adjusting the configuration of docker image by volume mounting files
+
+The simplest way of adding a new configuration file is adding it directly into `/etc/my.cnf.d` directory, like this.
+
+```
+docker run -d -v myconfig.cnf:/etc/my.cnf.d/myconfig.cnf <this_image>
```

Comment:
You will have to know how is the image structured and where are the appropriate folders first to do this, right? I mean the config file you're mounting has to know about the layout of the filesystem of the given image.

Also this won't work on Kubernetes/OpenShift if you don't provide a volume mount that is already populated with the config files that another container produced or you created by hand and uploaded to the volume.